tephra-core/src/main/java/org/apache/tephra/persist/AbstractTransactionLog.java

```diff
@@ -48,21 +52,30 @@
   protected long timestamp;
   private volatile boolean initialized;
   private volatile boolean closed;
+  private AtomicLong writtenUpTo = new AtomicLong();
   private AtomicLong syncedUpTo = new AtomicLong();
-  private List<Entry> pendingWrites = Lists.newLinkedList();
+  private Queue<Entry> pendingWrites = new ConcurrentLinkedQueue<>();
   private TransactionLogWriter writer;
 
-  public AbstractTransactionLog(long timestamp, MetricsCollector metricsCollector) {
+  private final AtomicLong positionBeforeAppend = new AtomicLong(0);
```

Comment:
This can be just a `long` field, right? Since it is only used by the start/stop timer method, which only gets called inside the writer sync block.

Reply:
oh yes, after the refactoring in the second commit it does not need to be atomic